packages/xo-web/src/xo-app/self/index.js

```diff
@@ -696,12 +700,26 @@ class ResourceSet extends Component {
 
 const compareName = (a, b) => (a.name < b.name ? -1 : 1)
 
+@addSubscriptions({ resourceSets: subscribeResourceSets })
+@connectStore({ areObjectsFetched })
 export default class Self extends Component {
   constructor (props) {
     super(props)
     this.state = {}
   }
 
+  componentWillReceiveProps (props) {
+    const { areObjectsFetched, resourceSets } = props
+    if (
+      areObjectsFetched &&
+      areObjectsFetched !== this.props.areObjectsFetched
+    ) {
+      this.setState({
+        resourceSets: resolveResourceSets(resourceSets).sort(compareName),
+      })
+    }
+  }
+
   componentWillMount () {
     this.componentWillUnmount = subscribeResourceSets(resourceSets => {
```

Comment:
If you subscribe to resource sets with `addSubscriptions`, you don't need this anymore.